### PR TITLE
September updates - deprecate `connector_ids`, parametrize constants

### PR DESCRIPTION
**Pull request by ibnt1**
Deprecate `connector_ids`, use and update `connectors` instead.
Fixes for no-split-at-connectors scenario.
Add parameters object to replace constants.

Keeping logic to calculate lrs for connectors for now instead of using values from the connectors object because:
- need to confirm if we still have edge cases where the coordinates of connectors are not found in the segment coordinates so that we can deprecate the fallback logic 
- need more work to add lrs to unit test data and confirm they are still capturing the intended edge cases they were targeting, and/or revisit the strategy for the tests

### CODE REVIEW

File: transportation_splitter.py [hunk 1]
```diff
@@ -331,50 +361,50 @@ def apply_lr_scope(
...
+    trs_to_keep: list[dict] = []
     for tr in turn_restrictions:
         tr_heading = (tr.get("when") or {}).get("heading")
         tr_sequence = tr.get("sequence")
         if not tr_sequence or len(tr_sequence) == 0:
             continue
 
-        if not connector_ids or len(connector_ids) != 2:
+        if not connectors or len(connectors) != 2:
             # at this point modified segments are expected to have exactly two connector ids, skip edge cases that don't
```
Comment (brad-richardson) [inline transportation_splitter.py:393]: We should check this early in the process - if turn restrictions are included in the config, split at connectors must be enabled as well

Comment (brad-richardson) [reply transportation_splitter.py:393]: Could add checking for valid configs as a future feature/issue if you want

File: transportation_splitter.py [hunk 2]
```diff
@@ -7,34 +7,73 @@
...
+    New split points are needed for linear references. This controls how far in meters from 
+    other existing splits (from either connectors or other LRs) do these LRs need to be 
+    for us to create a new connector for them instead of using the existing connector.
+    """
+    lr_split_point_min_dist_meters: float = 0.01 # 1cm
+
+    """
+    Skips steps for which intermediate streams are found, default True, set to False to always force reprocess all sub-steps 
+    """
+    reuse_existing_intermediate_outputs: bool = True
```
Comment (brad-richardson) [inline transportation_splitter.py:68]: Do you think it's feasible to run this without materializing intermediate steps? It would run a lot faster overall

File: transportation_splitter.py [hunk 3]
```diff
@@ -436,7 +466,20 @@ def get_length_bucket(length):
...
+    return {"connector_id": connector_id, "at": lr}
+
+def get_connectors_for_split(split_segment: SplitSegment, original_connectors: list[dict], original_segment_length:float) -> list[dict]:
+    connectors_for_split: list[dict] = [get_connector_dict(p.id, p.lr) for p in [split_segment.start_split_point, split_segment.end_split_point]]
+    connectors_for_split += [c for c in original_connectors if c["at"] > split_segment.start_split_point.lr and c["at"] < split_segment.end_split_point.lr]
+    connectors_for_split = sorted(connectors_for_split, key=lambda c: c["at"])    
+    # now recalculate the "at" location references to be relative to the split
+    for c in connectors_for_split:
+        if c["at"] is not None:
+            c["at"] = (c["at"] * original_segment_length - split_segment.start_split_point.lr_meters) / split_segment.length
```
Comment (brad-richardson) [inline transportation_splitter.py:479]: This is one big drawback to us including computed values like this, downstream processing has to understand the logic and re-implement

**Review by brad-richardson — APPROVED** [review]
